pkg/virt-handler/mount-manager/utils/recorder_test.go

```diff
@@ -0,0 +1,212 @@
+			Expect(err).ToNot(HaveOccurred())
+			if !equality.Semantic.DeepEqual(expectedRecord, res) {
+				ginkgo.Fail(fmt.Sprintf("expectedRecord %v not equal to %v", *expectedRecord, *res))
+			}
+		},
+			Entry("no cached values", false),
+			Entry("cached values", true),
+		)
+
+		DescribeTable("SetMountRecord", func(existingCachedValues bool) {
+			recordFile := filepath.Join(tempDir, string(vmi.UID))
+			expectedRecord := &vmiMountTargetRecord{UsesSafePaths: true}
+			if existingCachedValues {
+				writeRecordFile(recordFile, record.MountTargetEntries)
+				expectedRecord = record
+			}
+			newRecord := []MountTargetEntry{
+				{
+					TargetFile: "/test/target2",
+				},
+			}
+			err := m.SetMountRecord(vmi, newRecord)
+			Expect(err).ToNot(HaveOccurred())
+			expectedRecord.MountTargetEntries = newRecord
+			res, err := readRecordFile(recordFile)
+			Expect(err).ToNot(HaveOccurred())
+			if !equality.Semantic.DeepEqual(expectedRecord, res) {
+				ginkgo.Fail(fmt.Sprintf("expectedRecord %v not equal to %v", *expectedRecord, *res))
+			}
+		},
+			Entry("no cached values", false),
+			Entry("cached values", true),
+		)
+	})
+
+	Context("Clean-up", func() {
+		BeforeEach(func() {
+			tempDir = GinkgoT().TempDir()
+			m = NewMountRecorder(tempDir)
+			vmi = api.NewMinimalVMI("fake-vmi")
+			vmi.UID = "1234"
+		})
+
+		createTargetFiles := func(cd []MountTargetEntry) {
+			for _, entry := range cd {
+				file, err := os.Create(entry.TargetFile)
+				Expect(err).ToNot(HaveOccurred())
+				defer file.Close()
+				file, err = os.Create(entry.SocketFile)
+				Expect(err).ToNot(HaveOccurred())
+				defer file.Close()
+			}
+		}
+
+		areTargetFilesDeleted := func(cd []MountTargetEntry) bool {
+			for _, entry := range cd {
+				if _, err := os.Stat(entry.TargetFile); err == nil || !errors.Is(err, os.ErrNotExist) {
+					return false
+				}
+				if _, err := os.Stat(entry.SocketFile); err == nil || !errors.Is(err, os.ErrNotExist) {
+					return false
+				}
+			}
+			return true
+		}
+
+		fileNotExist := func(file string) bool {
+			_, err := os.Stat(file)
+			return errors.Is(err, os.ErrNotExist)
+		}
+
+		It("Should delete target files and record entry", func() {
+			cds := []MountTargetEntry{
+				{
+					TargetFile: filepath.Join(tempDir, "target0"),
+					SocketFile: filepath.Join(tempDir, "socket0"),
+				},
+				{
+					TargetFile: filepath.Join(tempDir, "target1"),
+					SocketFile: filepath.Join(tempDir, "socket1"),
+				},
+			}
+			createTargetFiles(cds)
+			m.SetMountRecord(vmi, cds)
+
+			err := m.DeleteMountRecord(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check clean-up
+			res, err := m.GetMountRecord(vmi)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(res).To(HaveLen(0))
+			Expect(areTargetFilesDeleted(cds)).To(BeTrue())
+			// No futher record the record file should be deleted
+			Expect(fileNotExist(filepath.Join(tempDir, string(vmi.UID)))).To(BeTrue())
+		})
+	})
+
+	Context("Hoplugged volume entries", func() {
```

Comment:
Nit: "For hotplug volume" "should be able to re-add entries after deleting them"